src/common/libkvs/treeobj.c

```diff
@@ -148,14 +178,30 @@ json_t *treeobj_get_data (json_t *obj)
     return data;
 }
 
-int treeobj_decode_val (json_t *obj, void **dp, int *lp)
+const char *treeobj_get_symlink (const json_t *obj)
+{
+    const char *type;
+    const json_t *data;
+    const char *str;
+
+    if (treeobj_peek (obj, &type, &data) < 0
+            || strcmp (type, "symlink") != 0) {
+        errno = EINVAL;
+        return NULL;
+    }
+    if (!(str = json_string_value (data)))
+        return NULL;
+    return str;
+}
```

Comment:
Very minor, but any reason not to just
```C
return (json_string_value (data));
```
? (thought perhaps you meant to set an errno here)

Reply:
Yeah I think that should set errno = EINVAL.  Thanks!

Reply:
Doh, I had forgotten the EINVAL part.  Thanks for the catch.